share/wake/lib/core/result.wake

```diff
@@ -107,6 +107,32 @@ export def rmapFail (fn: a => Result pass b): Result pass a => Result pass b = m
     Pass a = Pass a
     Fail f = fn f
 
+# Attempt to combine the elements of a `List` front-to-back into a single value.
+# An accumulator is updated from its initial value by combiningFn for each
+# element, and if any such update fails, the error value is returned instead.
+export def rfoldl (combiningFn: accumulator => element => Result accumulator error): accumulator => List element => Result accumulator error =
```

Comment:
If adding new functions to the standard library, please follow the style guide for comments (see: `map` or `unreachable` for examples to copy)